src/common/io.ts

```diff
@@ -154,38 +154,52 @@ export class ARIOReadable implements AoARIORead {
     });
   }
 
+  private async computeEpochIndexForTimestamp(
+    timestamp: number,
+  ): Promise<number> {
+    const epochSettings = await this.getEpochSettings();
+    const epochZeroStartTimestamp = epochSettings.epochZeroStartTimestamp;
+    const epochLengthMs = epochSettings.durationMs;
+    return Math.floor((timestamp - epochZeroStartTimestamp) / epochLengthMs);
+  }
+
+  private async epochInputToTags(
```

Comment:
nit: can return just one tag for no concat

Reply:
gets complex cause we need to type check undefined, empty array is simpler. think its fine as is